File: vdp-java/src/test/java/com/visa/vdp/visacheckout/TestGetPaymentData.java
package com.visa.vdp.visacheckout;

import java.io.IOException;
import java.security.KeyManagementException;
import java.security.KeyStoreException;
import java.security.NoSuchAlgorithmException;
import java.security.SignatureException;
import java.security.UnrecoverableKeyException;
import java.security.cert.CertificateException;

import org.apache.commons.lang3.StringUtils;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import com.visa.vdp.utils.AbstractClient;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestGetPaymentData extends AbstractClient {

	 String apiKey;
	 
	 @BeforeTest(groups = "visacheckout")
	 public void setup() {
		 apiKey = VisaProperties.getProperty(Property.API_KEY);
	 }
	 
	 @Test(groups = "visacheckout")
	 public void testGetPaymentInfo() throws SignatureException, IOException, KeyManagementException, UnrecoverableKeyException, NoSuchAlgorithmException, KeyStoreException, CertificateException {
	     String baseUri = "wallet-services-web/";
	     String resourcePath = "payment/data/{callId}";
	     resourcePath = StringUtils.replace(resourcePath, "{callId}", VisaProperties.getProperty(Property.CHECKOUT_CALL_ID));
	     
	     CloseableHttpResponse response = doXPayTokenGetRequest(baseUri, resourcePath, "apikey=" + this.apiKey, "Get Payment Information Test");
	     Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
	     response.close();
	 }
}

File: vdp-python/visa/test/visacheckout/test_get_payment_data.py
from visa.test.helpers.vdp_client_utils import VDPTestCaseClient
'''
@author: visa
'''

class TestGetPaymentData(VDPTestCaseClient):

    def test_get_payment_info(self):
        base_uri = 'wallet-services-web/'
        resource_path = 'payment/data/{callId}'
        resource_path = resource_path.replace('{callId}', self.config.get('VDP','checkoutCallId'))
        query_string = 'apikey=' + self.config.get('VDP','apiKey')
        response = self.do_x_pay_request(base_uri, resource_path , query_string, '', 'Get Payments Information Test', 'get')
        self.assertEqual(str(response.status_code) ,"200" ,"Get Payments Information test failed")
        pass